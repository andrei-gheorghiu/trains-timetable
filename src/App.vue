<script lang="ts">
import { defineComponent } from "vue";
import TrainsTimetable from "@/components/TrainsTimetable.vue";
export default defineComponent({
  name: "App",
  components: {
    TrainsTimetable,
  },
});
</script>

<template>
  <header />
  <main>
    <trains-timetable />
  </main>
  <footer />
</template>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2c3e50;
}
</style>
