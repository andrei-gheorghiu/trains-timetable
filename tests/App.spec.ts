import { shallowMount } from "@vue/test-utils";
import App from "@/App.vue";
import { createTestingPinia } from "@pinia/testing";
import axios from "axios";
import VueAxios from "vue-axios";

describe("TrainsTimetable.vue", () => {
  let consoleSpy: any;
  let wrapper: any;

  beforeEach(() => {
    consoleSpy = jest.spyOn(console, "error");
    wrapper = shallowMount(App, {
      global: {
        plugins: [createTestingPinia(), [VueAxios, axios]],
      },
    });
  });

  afterEach(() => {
    expect(consoleSpy).not.toHaveBeenCalled();
  });

  it("should render without errors", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
