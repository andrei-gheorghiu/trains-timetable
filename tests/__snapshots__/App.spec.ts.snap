// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainsTimetable.vue should render without errors 1`] = `
"<header></header>
<main>
  <trains-timetable-stub></trains-timetable-stub>
</main>
<footer></footer>"
`;
